contract/proxy-oracle/soroban/scripts/budget_report.py

```
@@ -0,0 +1,207 @@
+def extract_test_results(output: str) -> dict[str, str]:
+    """Parse cargo test output into {test_name: 'ok' | 'FAILED'}."""
+    results: dict[str, str] = {}
+    for line in output.splitlines():
+        m = re.search(r"test (\S+) \.\.\. (ok|FAILED)", line)
+        if m:
+            results[m.group(1)] = m.group(2)
+    return results
+
+
+def map_scenario(
+    scenario_label: str,
```

Ruff (0.15.15)

[warning] Unused function argument: `scenario_label`

(ARG001)